src/lib/expression/between_expression.cpp

```diff
@@ -0,0 +1,33 @@
+#include "between_expression.hpp"
+
+#include <sstream>
+
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+BetweenExpression::BetweenExpression(const std::shared_ptr<AbstractExpression>& value,
+                                     const std::shared_ptr<AbstractExpression>& lower_bound,
+                                     const std::shared_ptr<AbstractExpression>& upper_bound)
+    : AbstractPredicateExpression(PredicateCondition::Between, {value, lower_bound, upper_bound}) {}
```

Comment:
Do we want to assert that `lower <= upper`? tbh, I am not sure how SQL handles this. Does it invert the values?

Reply:
we can't assert that here, lower is just an expression, could refer to e.g., columns. at runtime, between is translated to `value >= lower AND value <= upper`

Reply:
`         6) "X BETWEEN Y AND Z" is equivalent to "X>=Y AND X<=Z".`